implementations/rust/ockam/ockam_entity/src/entity.rs

```diff
@@ -0,0 +1,363 @@
+use crate::{
+    Contact, ContactsDb, KeyAttributes, ProfileAuth, ProfileChangeEvent, ProfileChanges,
+    ProfileContacts, ProfileEventAttributes, ProfileIdentifier, ProfileIdentity, ProfileImpl,
+    ProfileSecrets, ProfileTrait, ProfileVault, ProfileVaultAccess,
+};
+
+use crate::EntityError::ProfileNotFound;
+use ockam_core::Result;
+use ockam_vault::ockam_vault_core::{PublicKey, Secret};
+
+/// An Entity represents an identity in various authentication contexts.
+#[derive(Clone)]
+pub struct Entity<V: ProfileVault> {
+    default_profile_identifier: ProfileIdentifier,
+    profiles: Vec<ProfileImpl<V>>,
+}
+
+impl<V: ProfileVault> Entity<V> {
+    #[allow(dead_code)]
+    pub fn new(default_profile: ProfileImpl<V>) -> Self {
+        let idref = default_profile.identifier();
+        let default_profile_identifier = ProfileIdentifier::from_key_id(idref.key_id().clone());
+        let profiles = vec![default_profile];
+        Entity {
+            default_profile_identifier,
+            profiles,
+        }
+    }
+
+    fn default_profile(&self) -> Option<&ProfileImpl<V>> {
```

Comment:
This way of getting the default profile can be used when only immutable access is needed